docs: re-add `studio document validation` and change code snippet

animeshk923 opened the pull request:
**Description**

- Re-added the `studio document validation` in the next-major-spec branch.
- Changed the code snippet to match v3.

**Related issue(s)**
Fixes #1862

animeshk923 committed:
re-add `studio document validation`

changes:

- Re-added the `studio document validation` in
the next-major-spec branch.
- Changed the code snippet to match v3.

animeshk923 committed:
changed versions in the required locations

derberg commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,139 @@
+
+If you did not follow the previous tutorial and do not have an `asyncapi.yaml` file ready, then generate one using `asyncapi new --example=tutorial.yml --no-tty` command.
+
+</Remember>
```
I don't think this should be here in this tutorial as this tutorial is about validating invalid document in studio, basing on example provided in studio - and command under `remember` note talks about generating a valid document

animeshk923 replied:
@derberg 
okay, so if the command generates a valid document, it should not be there. Any command to generate and invalid one? If not there is a way to generate one from AsyncAPI Studio as far as I've tested. Thoughts? :eyes:

derberg replied:
nah, I will just remove it

derberg commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,139 @@
+  onLightMeasured:
+    action: 'receive'
+    summary: Inform about environmental lighting conditions for a particular streetlight.
+    operationId: onLightMeasured
```
`operationId` is no longer there, the name of the key that describes an operation is the same as operationId

animeshk923 replied:
> `operationId` is no longer there, the name of the key that describes an operation is the same as operationId

Understood. I'm not much familiar with the structure of the file :sweat_smile: 

I guess it was resolved with a1bb873 am I correct?

derberg replied:
yup

derberg commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,139 @@
+    summary: Inform about environmental lighting conditions for a particular streetlight.
+    operationId: onLightMeasured
+    channel:
+      $ref: '#/channels/lightMeasured'
```
```suggestion
asyncapi: 3.0.0
info:
  title: Streetlights API
  version: '1.0.0'
  description: |
    The Smartylighting Streetlights API allows you
    to remotely manage the city lights.
  license:
    name: Apache 2.0
    url: 'https://www.apache.org/licenses/LICENSE-2.0'

servers:
  mosquitto:
    url: test.mosquitto.org
    protocol: mqtt

channels:
  lightMeasured:
    address: 'light/measured'
    messages:
      lightMeasuredMessage:
        name: LightMeasured
        payload:
          type: object
          properties:
            id:
              type: integer
              minimum: true
              description: Id of the streetlight.
            lumens:
              type: integer
              minimum: 0
              description: Light intensity measured in lumens.
            sentAt:
              type: string
              format: date-time
              description: Date and time when the message was sent.

operations:
  onLightMeasured:
    action: 'receive'
    summary: Inform about environmental lighting conditions for a particular streetlight.
    channel:
      $ref: '#/channels/lightMeasured'
```

I suggest we have an error focused not on `asyncapi` property (as with previous versions it was good as all v2 versions were super similar. In v3 things are different, and the fact that someone would do `asyncapi: 1.0.0` with v3 document seems like unrealistic error

thus I suggest error is that someone put `url` instead of `host`

-> later instructions also commented to match such case

animeshk923 replied:
resolved with a1bb873

derberg commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,139 @@
+ 
+1. You can see the error message on the screen: `Empty or invalid document. Please fix errors/define AsyncAPI document.`
+
+2. Open diagnostics, you can see more information related to your errors.
```
```suggestion
2. Open **DIAGNOSTICS**, you can see more information related to your errors.
```

this way it is clear that it refers to some tab/button/label in studio

animeshk923 replied:
Understood. Resolved with f1bc190

derberg commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,139 @@
+
+2. Open diagnostics, you can see more information related to your errors.
+
+3. Fix the incorrect AsyncAPI specification number to `3.0.0`.
```
```suggestion
3. Fix the incorrect server information. Use `host` instead of `url`
```

animeshk923 replied:
resolved with 05ef63a

derberg commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,139 @@
+asyncapi: '3.0.0'
+info:
+  title: Account Service
+  version: 1.0.0
```
```suggestion
servers:
  mosquitto:
    host: test.mosquitto.org
    protocol: mqtt
```

animeshk923 replied:
resolved with c5a7870

derberg commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,139 @@
+Notice how <b>description</b> property is missing; that doesn't make the AsyncAPI document invalid, but it's always better to include.
+</Remember>
+
+4. Read the next error: `must be number`. Fix the `minimum` by changing it to: `0`.
```
```suggestion
4. Read the next error: `"minimum" property type must be number`. Fix the `minimum` by changing it to: `0`.
```

animeshk923 replied:
resolved with 7d725ec

derberg commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,139 @@
+This tutorial taught us how to validate an AsyncAPI document using the AsyncAPI Studio tool. We also learned to troubleshoot an invalid AsyncAPI document by following the error message directions in diagnostics. In doing so, we learned how to identify `REQUIRED` properties in all AsyncAPI documents.
+
+## Next steps
+Now that you have completed this tutorial, go ahead to learn [generate AsyncAPI messages (events)](https://asyncapi.com/docs/tutorials/generate-code) which you will be sending to your application.
```
```suggestion
Now that you have completed this tutorial, go ahead to learn [generate AsyncAPI messages (events)](/docs/tutorials/generate-code) which you will be sending to your application.
```

we should always use relative links

animeshk923 replied:
> we should always use relative links

Understood.

animeshk923 replied:
resolved with 27a0b69

derberg commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,139 @@
+## Next steps
+Now that you have completed this tutorial, go ahead to learn [generate AsyncAPI messages (events)](https://asyncapi.com/docs/tutorials/generate-code) which you will be sending to your application.
+
+You may also enjoy reading our [AsyncAPI document validation guide](https://asyncapi.com/docs/guides/validate).
```
```suggestion
You may also enjoy reading our [AsyncAPI document validation guide](/docs/guides/validate).
```

animeshk923 replied:
relative links, understood this.

animeshk923 replied:
resolved with 7a13c1b

derberg commented on pages/docs/tutorials/studio-document-validation.md:
````diff
@@ -0,0 +1,139 @@
+              description: Date and time when the message was sent.
+```
+
+6. Congratulations! You identified and fixed all the errors, and now have a valid AsyncAPI document.
````
```suggestion
5. Congratulations! You identified and fixed all the errors, and now have a valid AsyncAPI document.
```

yeah, I don't think we need to showcase another error with schema, especially that these errors are a bit unclear

animeshk923 replied:
hmm...this is a bit technical so you would know the best :sweat_smile:

animeshk923 replied:
resolved with a4beea4

derberg reviewed (commented):
@AnimeshKumar923 made some suggestions

animeshk923 committed:
Update pages/docs/tutorials/studio-document-validation.md

applied suggestion from: https://github.com/asyncapi/website/pull/2276#discussion_r1386853405

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

animeshk923 committed:
Update pages/docs/tutorials/studio-document-validation.md

applied suggstion from: https://github.com/asyncapi/website/pull/2276#discussion_r1386855249

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

animeshk923 committed:
Update pages/docs/tutorials/studio-document-validation.md

applied suggestion from: https://github.com/asyncapi/website/pull/2276#discussion_r1386856435

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

animeshk923 committed:
Update pages/docs/tutorials/studio-document-validation.md

applied suggestion from: https://github.com/asyncapi/website/pull/2276#discussion_r1386856975

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

animeshk923 committed:
Update pages/docs/tutorials/studio-document-validation.md

applied suggestion from: https://github.com/asyncapi/website/pull/2276#discussion_r1386859457

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

animeshk923 committed:
Update pages/docs/tutorials/studio-document-validation.md

applied suggestion from: https://github.com/asyncapi/website/pull/2276#discussion_r1386861354

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

animeshk923 committed:
Update pages/docs/tutorials/studio-document-validation.md

applied suggestion from: https://github.com/asyncapi/website/pull/2276#discussion_r1386861621

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

animeshk923 committed:
Update pages/docs/tutorials/studio-document-validation.md

applied suggestion from: https://github.com/asyncapi/website/pull/2276/files#r1386867168

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

derberg committed:
Update studio-document-validation.md

quetzalliwrites commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,116 @@
+---
+title: "Validate AsyncAPI document with Studio"
+description: This tutorial will teach you how to validate AsyncAPI documents using the AsyncAPI Studio tool.
```
```suggestion
description: In this tutorial, you'll learn how to validate AsyncAPI documents using the AsyncAPI Studio tool.
```

quetzalliwrites commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,116 @@
+---
+
+## Introduction
+This tutorial will teach you how to validate AsyncAPI documents using the [AsyncAPI Studio tool](https://studio.asyncapi.com/).
```
```suggestion
In this tutorial, you'll learn how to validate AsyncAPI documents using the [AsyncAPI Studio tool](https://studio.asyncapi.com/).
```

quetzalliwrites commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,116 @@
+5. Congratulations! You identified and fixed all the errors, and now have a valid AsyncAPI document.
+
+## Summary
+This tutorial taught us how to validate an AsyncAPI document using the AsyncAPI Studio tool. We also learned to troubleshoot an invalid AsyncAPI document by following the error message directions in diagnostics. In doing so, we learned how to identify `REQUIRED` properties in all AsyncAPI documents.
```
```suggestion
In this tutorial, you learned how to validate an AsyncAPI document using the AsyncAPI Studio tool. You also learned to troubleshoot an invalid AsyncAPI document by following the error message directions in diagnostics. In doing so, you learned how to identify `REQUIRED` properties in all AsyncAPI documents.
```

quetzalliwrites commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,116 @@
+## Introduction
+This tutorial will teach you how to validate AsyncAPI documents using the [AsyncAPI Studio tool](https://studio.asyncapi.com/).
+
+You will start with a broken AsyncAPI document and troubleshoot via console errors step-by-step until we end up with a valid AsyncAPI document. This process will illustrate how to identify [`REQUIRED` properties in AsyncAPI documents](https://www.asyncapi.com/docs/reference/specification/latest#A2SObject).
```
```suggestion
You will start with a broken AsyncAPI document and troubleshoot via console errors step-by-step until you end up with a valid AsyncAPI document. This process will illustrate identifying [`REQUIRED` properties in AsyncAPI documents](https://www.asyncapi.com/docs/reference/specification/latest#A2SObject).
```

quetzalliwrites commented on pages/docs/tutorials/studio-document-validation.md:
```diff
@@ -0,0 +1,116 @@
+Now let's experiment with an invalid file to see how errors are displayed and how to make that file valid again.
+
+## Copy invalid AsyncAPI document
+Let's pretend we have an invalid AsyncAPI document.
```
```suggestion
Let's pretend you have an invalid AsyncAPI document.
```

quetzalliwrites commented on pages/docs/tutorials/studio-document-validation.md:
````diff
@@ -0,0 +1,116 @@
+``` 
+ 
+## Troubleshoot Studio errors
+Let's fix the errors one by one until we end up with a valid AsyncAPI document.
````
```suggestion
Let's fix the errors one by one until you end up with a valid AsyncAPI document.
```

Commit:
fixed 'we' instances and grammar issues

quetzalliwrites commented:
/rtm
